cc3d/CompuCellSetup/SimulationThread.py

```diff
@@ -0,0 +1,127 @@
+from cc3d.core.enums import SimType
+
+
+class SimulationThread:
+    """
+    A minimal class for hooking a custom simulation controller into the CC3D core
+
+    For reference to hooking into the CC3D core when designing a derived class,
+    see cc3d.CompuCellSetup.simulation_setup.py
+    """
+
+    sim_type = SimType.THREADED
+
+    def __init__(self):
+        pass
+
+    def inject(self):
+        """
+        Inject this into the CC3D core
+
+        :return: None
+        """
+        from cc3d.CompuCellSetup import persistent_globals
+        persistent_globals.simthread = self
+        persistent_globals.sim_type = self.sim_type
+
+    @staticmethod
+    def main_loop():
+        """
+        Sequence to control what occurs during a run of a simulation
+
+        If this is not overloadeded, then CC3D will function out-of-the-box
+
+        If CC3D detects this in persistent_globals.simthread, then it will use what's returned by this function to
+            perform the simulation
+
+        The returned function must have the form
+            def f(CompuCell.Simulator, SimulationThread, CompuCell.SteppableRegistry) -> None
+
+        Note that persistent_globals.player_type is an optional global variable to use for conditionalizing which
+            function is returned here
+
+        See cc3d.CompuCellSetup.simulation_setup.main_loop for an example of how to set up an automated simulation run
+
+        :return: None
+        """
+        from cc3d.CompuCellSetup import main_loop
```

Comment:
do we want this import here or at the top?

Reply:
I found placement here to be necessary to support implementations by external players, due to downstream declarations of global variables.

Reply:
I thought it was something like this. Let's leave it for now and then we can work on writing "smart" import headers

Reply:
Works for me.